python_legacy/iceberg/core/base_table_scan.py

```diff
@@ -125,6 +125,26 @@ def option(self, property, value):
                                      selected_columns=self.selected_columns, options=builder,
                                      minused_cols=self.minused_cols)
 
+    def check_partition_exist(self, ops=None, snapshot=None, row_filter=None):
+
+        if not all(i is None for i in [ops, snapshot, row_filter]):
+            raise NotImplementedError()
+
+        snapshot = self.ops.current().snapshot(self.snapshot_id) \
+            if self.snapshot_id is not None else self.ops.current().current_snapshot()
+
+        if snapshot is not None:
+            _logger.info("Checking partitions for table {} snapshot {} created at {} with filter {}"
```

Comment:
Do we need to log filter. In some cases filter may contain sensitive information that we should not log.